### PR TITLE
chore: Use lifecycle for arguments deprecation in ggpairs

### LINKED ISSUE #365: Use lifecycle to issue deprecation warnings
See https://lifecycle.r-lib.org/articles/lifecycle.html#deprecating-arguments

### PR DESCRIPTION
**Pull request by 92amartins**
This aims to partially fix #365.

It only deprecates arguments in the `ggpairs`.

If desired I can follow up with deprecation warnings for the other files that have functions deprecated functions/arguments:

- `deprecated.R`
- `gg-plots.R`
- `gg_matrix_gtable.R`
- `ggnet.R`

### CODE REVIEW

File: R/ggpairs.R [hunk 1]
```diff
@@ -623,8 +608,8 @@ ggduo <- function(
 #' @param upper see Details
 #' @param lower see Details
 #' @param diag see Details
-#' @param params deprecated.  Please see \code{\link{wrap_fn_with_param_arg}}
-#' @param ... deprecated. Please use \code{mapping}
+#' @param params `r lifecycle::badge("deprecated")` see \code{\link{wrap_fn_with_param_arg}}
```
Comment (schloerke) [inline R/ggpairs.R:611]:
```suggestion
#' @param params `r lifecycle::badge("deprecated")` See \code{\link{wrap_fn_with_param_arg}}
```

File: R/ggpairs.R [hunk 2]
```diff
@@ -623,8 +608,8 @@ ggduo <- function(
...
+#' @param ... `r lifecycle::badge("deprecated")` use \code{mapping}
```
Comment (schloerke) [inline R/ggpairs.R:612]:
```suggestion
#' @param ... `r lifecycle::badge("deprecated")` Use \code{mapping}
```

File: R/ggpairs.R [hunk 3]
```diff
@@ -781,10 +766,27 @@ ggpairs <- function(
...
+      when = "2.2.2",
+      what = "ggpairs(params)"
+    )
+  }
+  if(lifecycle::is_present(list(...))){
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(...)"
+    )
+  }
```
Comment (schloerke) [inline R/ggpairs.R:788]: Could we use `rlang::check_dots_empty()`?

Comment (schloerke) [reply R/ggpairs.R:788]: This would require rlang >= 1.0.0

**Review by schloerke — CHANGES_REQUESTED** [review]
Please accept new snaps or remove the new gglegend snaps

**Comment by schloerke** [comment]
> If desired I can follow up with deprecation warnings for the other files that have functions deprecated functions/arguments:
> 
> * `deprecated.R`
> * `gg-plots.R`
> * `gg_matrix_gtable.R`
> * `ggnet.R`

Yes, please! Thank you

File: R/ggpairs.R (hunk 3 above)
Comment (92amartins) [reply R/ggpairs.R]: Yes, I guess we can, but I needed to tweak it a bit in order to not throw an error and keep the function without breaking changes for the users.

1fc3d59f6ae750e2d261308fb89068cbe911d9f7

**Comment by 92amartins** [comment]
> > If desired I can follow up with deprecation warnings for the other files that have functions deprecated functions/arguments:
> > 
> > * `deprecated.R`
> > * `gg-plots.R`
> > * `gg_matrix_gtable.R`
> > * `ggnet.R`
> 
> Yes, please! Thank you

I'm thinking about breaking down these changes in maybe 2 or 3 PR's, what do you think?

File: R/ggpairs.R (hunk 3 above)
Comment (schloerke) [reply R/ggpairs.R]: Thank you.

`lifecycle::is_present(list())` is always `TRUE`. Now we're able to check if `...` exists

File: R/ggpairs.R [hunk 4]
```diff
@@ -456,8 +435,14 @@ ggduo <- function(
     progress = NULL,
     xProportions = NULL,
     yProportions = NULL,
-    legends = stop("deprecated")) {
-  warn_deprecated(!missing(legends), "legends")
+    legends = deprecated()) {
+  if(lifecycle::is_present(legends)){
```
Comment (schloerke) [inline R/ggpairs.R:439]:
```suggestion
  if (lifecycle::is_present(legends)) {
```

File: R/ggpairs.R [hunk 5]
```diff
@@ -781,10 +766,24 @@ ggpairs <- function(
     cardinality_threshold = 15,
     progress = NULL,
     proportions = NULL,
-    legends = stop("deprecated")) {
-  warn_deprecated(!missing(legends), "legends")
-  warn_if_args_exist(list(...))
-  stop_if_params_exist(params)
+    legends = deprecated()) {
+  if(lifecycle::is_present(legends)){
```
Comment (schloerke) [inline R/ggpairs.R:770]:
```suggestion
  if (lifecycle::is_present(legends)) {
```

File: R/ggpairs.R [hunk 6]
```diff
@@ -781,10 +766,24 @@ ggpairs <- function(
...
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(legends)",
+      details = "Ability to put legends in each plot will be dropped in next releases."
+    )
+  }
+  if(lifecycle::is_present(params)){
```
Comment (schloerke) [inline R/ggpairs.R:777]:
```suggestion
  if (lifecycle::is_present(params)) {
```

File: R/ggpairs.R [hunk 7]
```diff
@@ -781,10 +766,24 @@ ggpairs <- function(
...
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(params)"
+    )
+  }
+  has_dots = rlang::check_dots_empty(error = function(cnd) TRUE)
+  if(isTRUE(has_dots)){
```
Comment (schloerke) [inline R/ggpairs.R:784]:
```suggestion
  if (isTRUE(has_dots)) {
```

File: R/ggpairs.R [hunk 8]
```diff
@@ -781,10 +766,24 @@ ggpairs <- function(
...
+      details = "Ability to put legends in each plot will be dropped in next releases."
+    )
+  }
+  if(lifecycle::is_present(params)){
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(params)"
+    )
+  }
+  has_dots = rlang::check_dots_empty(error = function(cnd) TRUE)
```
Comment (schloerke) [inline R/ggpairs.R:783]:
```suggestion
  has_dots = rlang::check_dots_empty(error = function(cnd) {TRUE})
```

**Comment by schloerke** [comment]
> I'm thinking about breaking down these changes in maybe 2 or 3 PR's, what do you think?

Let's get this PR merged in so that your GHA runs will happen automatically. If it's all in the same idea... let's do it in a single followup PR.

File: R/ggpairs.R [hunk 9]
```diff
@@ -781,10 +766,24 @@ ggpairs <- function(
...
+      details = "Ability to put legends in each plot will be dropped in next releases."
+    )
+  }
+  if (lifecycle::is_present(params)) {
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(params)"
+    )
+  }
+  has_dots = rlang::check_dots_empty(error = function(cnd) {TRUE})
```
Comment (schloerke) [inline R/ggpairs.R:783]: lints
```suggestion
  has_dots <- rlang::check_dots_empty(error = function(cnd) {
    TRUE
  })
```

**Comment by schloerke** [comment]
Merging

The two ubuntu errors are due to install failures from a dep package for `{emmeans}` unrelated to this PR. (Happens on `master` branch.)

**Pull request merged**